### ISSUE TITLE
Latest version does not open any web page

### ISSUE THREAD

**Issue by lowtrucks** [bromite/bromite#216]
**Bromite version**
Version: `71.0.3578.120`
Arch: `arm`
Android version: `9.0 (LineageOS 16.0)`

**Is the bug reproducible with latest version?**
yes.

**Describe the bug**
Bromite doesn't open **any** Web page, even local _about:blank_, which doesn't require internet connection.

**To Reproduce**
run Bromite with any Web page.

**Expected behavior**
a browser opening any Web page.

**Screenshots**
![50758785-ffe9b480-1284-11e9-9e8e-9090225ea714](https://user-images.githubusercontent.com/12945674/50773260-3558c700-12b2-11e9-9ced-05887bc9e2ac.png)

**Comment by csagan5** [comment]
@lowtrucks it seems to be specific of ARM; I cannot reproduce this with x86 or ARM64.

Can you extract some Bromite crash from the logs (`adb logcat`)? I think it is crashing.

**Comment by meijkl** [comment]
@csagan5 : had the same issue on my Nexus 6p, which is arm64 based. Unfortunately I've already uninstalled Bromite Webview v120.

**Comment by Asbesbopispa** [comment]
> 
> 
> @lowtrucks it seems to be specific of ARM; I cannot reproduce this with x86 or ARM64.
> 
> Can you extract some Bromite crash from the logs (`adb logcat`)? I think it is crashing.

I tested a bit with my phone, and I discovered that's it's a Chromium related problem. I tried the latest Chromium System WebView build on my phone and many apps are crashing like the Bromite one. The latest Bromite Browser works normally instead.
I'm on Android 8.1 LineageOS 15.1

**Comment by felipe19930** [comment]
Crashing for me as well

Moto G falcon / LineageOS 14.1 microG / Bromite 71.0.3578.120

[adb-log-bromite-crash.txt](https://github.com/bromite/bromite/files/2733923/adb-log-bromite-crash.txt)

**Comment by Nickdv9** [comment]
Web view crashing here as well. I reopened an issue with some logs for arm64 here [#219](https://github.com/bromite/bromite/issues/219) if that helps.

**Comment by jenslody** [comment]
Bromite -120(browser not webview) prevents DuckDuckGo-browser and Jelly (webview based) from opening any webpage, and it blocks Magisk's grant-root-access-dialog from popping up.
It's currently not installed, but I can fetch a logcat later.

**Comment by csagan5** [comment]
@lowtrucks @Asbesbopispa @felipe19930 can you test with this pre-release version? https://github.com/bromite/bromite/releases/tag/71.0.3578.123

**Comment by lowtrucks** [comment]
i just tested **71.0.3578.123**. it still doesn't load any Web page, even _about:blank_.

**Comment by felipe19930** [comment]
It still crashes for me. But I'm wondering if it's mandatory to also update the WebView, because I'm not sure how I do that:

`adb: failed to install arm_SystemWebView.apk: Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.android.webview signatures do not match the previously installed version; ignoring!]`

So I just upgraded Bromite using `arm_ChromeModernPublic.apk`. Anyway, here's the logcat:

[adb-logcat-2.txt](https://github.com/bromite/bromite/files/2734414/adb-logcat-2.txt)

**Comment by csagan5** [comment]
> i just tested **71.0.3578.123**. it still doesn't load any Web page, even _about:blank_.

Without the crash log I cannot troubleshoot this. I will use @felipe19930 crash log.

> But I'm wondering if it's mandatory to also update the WebView, because I'm not sure how I do that:
> 
> `adb: failed to install arm_SystemWebView.apk: Failure [INSTALL_FAILED_UPDATE_INCOMPATIBLE: Package com.android.webview signatures do not match the previously installed version; ignoring!]`

This issue here is about Bromite, for the SystemWebview there is #219. The issues are probably related but need to be troubleshot separately.

**Comment by csagan5** [comment]
The only possibility for this crash is this commit: https://github.com/chromium/chromium/commit/452ab6c02de3d54b9590c602a440c9fb3ae0c49e

There is a new build for .123 in progress that reverts that commit, as soon as the new build files appear here https://github.com/bromite/bromite/releases/tag/71.0.3578.123 @felipe19930 @lowtrucks @Asbesbopispa you can try to see if that solves it.

**Comment by biswasab** [comment]
I am affected as well. Bromite doesn't open any web pages at all(Samsung Galaxy j7 2017).

I am rolling back to the previous working version for the mean time and subscribing to this bug as well.

**Comment by csagan5** [comment]
@lowtrucks installing from scratch would not make a difference; I will make a new build, do you have the crash log? Only the part that looks like this is needed (lines with `F DEBUG`):
```
01-07 19:50:43.222  2203  2203 I Zygote  : Process 28705 exited due to signal (7)
01-07 19:50:43.296 28825 28825 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
01-07 19:50:43.298 28825 28825 F DEBUG   : LineageOS Version: 'xxx'
01-07 19:50:43.298 28825 28825 F DEBUG   : Build fingerprint: 'motorola/falcon_retbr_ds/falcon_umtsds:5.1/LPB23.13-56/58:user/release-keys'
01-07 19:50:43.299 28825 28825 F DEBUG   : Revision: '0'
01-07 19:50:43.299 28825 28825 F DEBUG   : ABI: 'arm'
01-07 19:50:43.299 28825 28825 F DEBUG   : pid: 28785, tid: 28804, name: CrUtilityMain  >>> org.bromite.bromite:sandboxed_process1 <<<
01-07 19:50:43.299 28825 28825 F DEBUG   : signal 7 (SIGBUS), code 1 (BUS_ADRALN), fault addr 0x9c0ee791
01-07 19:50:43.299 28825 28825 F DEBUG   :     r0 9bee3800  r1 9c0ee86c  r2 00000000  r3 00000001
01-07 19:50:43.299 28825 28825 F DEBUG   :     r4 50204030  r5 00000000  r6 9c3544e4  r7 00000000
01-07 19:50:43.299 28825 28825 F DEBUG   :     r8 9c0ee791  r9 9c0f794c  sl 00000000  fp 99321ac3
01-07 19:50:43.299 28825 28825 F DEBUG   :     ip 9c08cea8  sp 9c354490  lr 9a50c77f  pc 9a50c422  cpsr 800b0030
01-07 19:50:43.305 28825 28825 F DEBUG   : 
01-07 19:50:43.305 28825 28825 F DEBUG   : backtrace:
01-07 19:50:43.305 28825 28825 F DEBUG   :     #00 pc 00c3e422  /data/app/org.bromite.bromite-1/base.apk (offset 0x1843000)
01-07 19:50:43.359 28825 28825 F DEBUG   :     #01 pc 00c3e77b  /data/app/org.bromite.bromite-1/base.apk (offset 0x1843000)
```

**Comment by lowtrucks** [comment]
i included logcat from 12:39 to 12:41, when i was running Bromite, and it didn't load google.com site.

it doesn't crash, it just doesn't open any Web pages.

[log.txt](https://github.com/bromite/bromite/files/2735614/log.txt)

is this log enough?

**Comment by lowtrucks** [comment]
understoo, thanks. will be trying next build, as soon as it compiled.

**Comment by lowtrucks** [comment]
i just tested fresh 71.0.3578.123. it still doesn't work.

logcat:

[log2.txt](https://github.com/bromite/bromite/files/2736225/log2.txt)

**Comment by fitittome** [comment]
I just tested fresh 71.0.3578.123 build (arm_ChromeModernPublic) on a device running unofficial Android 9 (LineageOS 16.0) still the same.  Partial logcat attached.
[2019-01-08-10-51-36.txt](https://github.com/bromite/bromite/files/2736477/2019-01-08-10-51-36.txt)

**Comment by csagan5** [comment]
@fitittome in your logcat I do not see the crash dump with lines `F DEBUG`, is it possible they are suppressed or printed somewhere else in your log?

**Comment by fitittome** [comment]
> @fitittome in your logcat I do not see the crash dump with lines `F DEBUG`, is it possible they are suppressed or printed somewhere else in your log?

I noticed that.  Here you go, this is a fresh complete log.  It's a virgin test device so no privacy issues.
[2019-01-08-12-14-43.txt](https://github.com/bromite/bromite/files/2736668/2019-01-08-12-14-43.txt)

If it would be helpful I could reinstalled with Android 7 or 8, if that would assist.

**Comment by csagan5** [comment]
@fitittome no crash in there, I wonder if logging crashes is disabled somehow.

@lowtrucks @fitittome @felipe19930 does any of you also have the SystemWebView installed? (you don't have to install it, I am just inquiring to see if it's correlated)

**Comment by fitittome** [comment]
> @fitittome no crash in there, I wonder if logging crashes is disabled somehow.
> 
> @lowtrucks @fitittome @felipe19930 does any of you also have the SystemWebView installed? (you don't have to install it, I am just inquiring to see if it's correlated)

It's odd.  Its Matlog, recording the Main log on verbose.  If done the adb permission thing, it must be right given everything else it's recording in the log.    But I'm getting the Aw Snap page on loading Bromite.  I don't have SystemwebVeiw installed.  The in-built browser works fine.  I'm going to re-install Android 8 (Lineage 15.1) on the device and see what I get there.

**Comment by lowtrucks** [comment]
@csagan5, yes i have **Android System WebView** installed.

if i disabled it, Gmail stops working (it crashes).
![screen_20190108_1751](https://user-images.githubusercontent.com/12945674/50831932-61398280-136e-11e9-8879-a9e229f07ca8.png)

**Comment by csagan5** [comment]
> It's odd. Its Matlog, recording the Main log on verbose. If done the adb permission thing, it must be right given everything else it's recording in the log. But I'm getting the Aw Snap page on loading Bromite. I don't have SystemwebVeiw installed. The in-built browser works fine. I'm going to re-install Android 8 (Lineage 15.1) on the device and see what I get there.

Perhaps you would get the crashes if using `adb logcat` from a desktop?

@lowtrucks I meant the Bromite SystemWebView. it's fine, you don't need to change anything.

**Comment by felipe19930** [comment]
>The uploads are now complete; please let me know if situation changed.

I was going to test it, but I don't see any apks to download. Did they get deleted?

**Comment by lowtrucks** [comment]
@felipe19930, yes, because @csagan5 probably run another compile for next test builds.

**Comment by csagan5** [comment]
@felipe19930 yes, there is a new build that will appear there in a few hours. The one I deleted would probably crash for you too, as it seems that you guys all have the same scenario (although the crash logs are different sometimes).

So far I have not been able to reproduce it on ARM64 and emulated x86/ARM; when the next build is done and after you guys test it again, if it still crashes I would like to compare the crash log for everyone, to see at least whether it is the same crash or not.

**Comment by fitittome** [comment]
Curiously, I reinstalled 8 (LineageOS 15.1).  I downloaded the 120 Bromite as the 123 had gone.  Then captured the actual desktop adb logcat whist getting the 'Aw Snap'.  I don't know if it helps and whilst we are waiting for the rebuild.
[15_1_log.txt](https://github.com/bromite/bromite/files/2737195/15_1_log.txt)

**Comment by csagan5** [comment]
@fitittome no, the crash dumps look like this: https://github.com/bromite/bromite/issues/216#issuecomment-452206778

Your log mentions a crash but I cannot see it there, no idea why.

**Comment by csagan5** [comment]
On older Android (4.1) the crash log looks like this:
```
F/libc    ( 8510): Fatal signal 7 (SIGBUS) at 0x57a5b791 (code=1), thread 8527 (CrRendererMain)
I/DEBUG   ( 8535): handle_request(12)
I/DEBUG   ( 8535): BOOM: pid=8510 uid=99002 gid=99002 tid=8527
I/DEBUG   ( 8529): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   ( 8529): Build fingerprint: 'xxx:user/release-keys'
I/DEBUG   ( 8529): pid: 8497, tid: 8522, name: CrRendererMain  >>> org.bromite.bromite:sandboxed_process1 <<<
I/DEBUG   ( 8529): signal 7 (SIGBUS), code 1 (BUS_ADRALN), fault addr 57a5b791
I/DEBUG   ( 8535): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
I/DEBUG   ( 8535): Build fingerprint: 'xxx:user/release-keys'
I/DEBUG   ( 8535): pid: 8510, tid: 8527, name: CrRendererMain  >>> org.bromite.bromite:sandboxed_process2 <<<
I/DEBUG   ( 8535): signal 7 (SIGBUS), code 1 (BUS_ADRALN), fault addr 57a5b791
I/DEBUG   ( 8529):     r0 57850800  r1 57a5b86c  r2 00000001  r3 52184730
I/DEBUG   ( 8529):     r4 5f004030  r5 00000000  r6 535a474c  r7 00000000
I/DEBUG   ( 8529):     r8 57a5b791  r9 4fb1d358  sl 4fb1d340  fp 521a2338
I/DEBUG   ( 8529):     ip 00000000  sp 535a46f8  lr 55e7977f  pc 55e79422  cpsr 00000030
I/DEBUG   ( 8529): 
I/DEBUG   ( 8529): backtrace:
I/DEBUG   ( 8529):     #00  pc 00c3e422  /data/data/org.bromite.bromite/lib/libchrome.so
I/DEBUG   ( 8529):     #01  pc 00c3e77b  /data/data/org.bromite.bromite/lib/libchrome.so
I/DEBUG   ( 8529): 
I/DEBUG   ( 8529): stack:
I/DEBUG   ( 8529):          535a46b8  4fb20f98  
I/DEBUG   ( 8529):          535a46bc  55637d0d  /data/data/org.bromite.bromite/lib/libchrome.so
I/DEBUG   ( 8529):          535a46c0  535a46fc  [stack:8522]
```

**Comment by csagan5** [comment]
@lowtrucks @fitittome @felipe19930 I believe the problem is solved now; at least with the ARM device I used it cannot be reproduced anymore with latest build.

Can you guys confirm?

**Comment by lowtrucks** [comment]
@csagan5, fresh 71.0.3578.123 build (arm_ChromeModernPublic) works fine.

thanks for fixing the problem!

**Comment by csagan5** [comment]
@nikolowry as you can see from this issue and #219 using `use_jumbo_build=true` (from scratch) causes inexplicable crashes for ARM architecture; it is probably a bug worth reporting upstream. I remember I stopped using that flag (mentioned in https://github.com/nikolowry/bromite-builder/issues/1#issuecomment-450936785) but I had not noted down the reason.

See if you also can reproduce the crashes with your jumbo builds on ARM?

**Comment by felipe19930** [comment]
It stopped crashing for me as well, thanks a lot!

**Comment by csagan5** [comment]
Fixed in `71.0.3578.123`

**Comment by nikolowry** [comment]
@csagan5 my personal builds are always for arm64 -- which is why I never ran into this issue. The few times I did run partial arm builds I did see warnings about `target-abi-soft, sorry for not mentioning those

Regardless, I removed the hardcoded `use_jumbo_build` and instead now support passing custom GN args via CLI options.

**Comment by csagan5** [comment]
> The few times I did run partial arm builds I did see warnings about `target-abi-soft`, sorry for not mentioning those

I have not seen any such warning, do you have a sample?

> Regardless, I removed the hardcoded `use_jumbo_build`

I think it would have been more useful to verify that using such options creates crashing APKs also in your builds.

**Comment by nikolowry** [comment]
@csagan5 Either tonight or sometime over the weekend, I'll locally run `./bromite-builder --gn-args="use_jumbo_build=true"` and afterwards upload/link the APK here.  I'll also include any relevant build warnings as well.

**Comment by nikolowry** [comment]
@csagan5 haven't loaded up on a device or emulator yet (so logcat pending, unless you beat me to it), but here's the APK https://neeks.me/cloud/index.php/s/iFfKCQfYaK3TSw9.  The warning I was referring to before -- which was the only additional warning unique from my arm64 builds -- was:

```
[2273/22809] CC obj/third_party/zlib/zlib_arm_crc32/crc32_simd.o
'+soft-float-abi' is not a recognized feature for this target (ignoring feature)
'+soft-float-abi' is not a recognized feature for this target (ignoring feature)
```

**Comment by csagan5** [comment]
@nikolowry unfortunately I do not have a device to test that

**Comment by nikolowry** [comment]
@csagan5 I recently got a new machine and have been distracted trying to get it "just right" -- which means my Android dev environment isn't setup yet and I'm without an emulator.

I'm still on 71.0.3578.120, so my plan is to build/install the latest arm64 on my device . Following that I'll install the arm version and record the logcat and then I'll switch back to the arm64 build (don't feel like erasing user data, that's why I'm going this route

**Comment by csagan5** [comment]
@nikolowry it's fine, I don't think there is urgency or a necessary ETA. But if you were to find that your ARM build with jumbo does **not**crash, that would be interesting and worthy of further investigation...

**Comment by csagan5** [comment]
> I got nearly the same results with the build I linked previously (Aw snap vs Can't Find Page).

That is enough; the logcat does not contain the crash dump, probably you have developer mode not enabled or similar.

Now one can conclude that if upstream uses jumbo builds and they are not getting the crashes...then there is something wrong somewhere, but I have not the patience/will to follow this upstream (feel free if you like).

**Comment by MassilloAriel** [comment]
Hi!
I'm late to this thread, but I came across the same issue when installing the app from F-Droid on LineageOS.
For some reason, even on the latest version of LineageOS _(at the moment is 18.1 - Android 11)_ the app kept crashing.
I went back some versions of Bromite and even when trying to install the .apk manually the same behavior kept happening.
I managed to install it with adb: 

> adb install arm64_ChromePublic.apk

For some reason, the default LineageOS installer is causing this problem. I faced the same issue with other apps.
Hope this helps anyone.

**Comment by MassilloAriel** [comment]
Hi!
I'm late to this thread, but I came across the same issue when installing the app from F-Droid on LineageOS.
For some reason, even on the latest version of LineageOS _(at the moment is 18.1 - Android 11)_ the app kept crashing.
I went back some versions of Bromite and even when trying to install the .apk manually the same behavior kept happening.
Debug mode is not the issue, I tried installing the app with and without it.

Finally, I managed to install it with adb: 

> adb install arm64_ChromePublic.apk

For some reason, the LineageOS app installer is causing this problem. I faced the same issue with other apps.
Hope this helps anyone.
